Makefile

```diff
@@ -460,6 +460,7 @@ READELF		= $(CROSS_COMPILE)readelf
 STRIP		= $(CROSS_COMPILE)strip
 endif
 RUSTC		= rustc
+RUSTC_BOOTSTRAP = 1
```

Comment:
You can give the name of specific crates (separated with spaces I think). Rustc will then only allow unstable features for those crates. This could be used as a roadblock to prevent depending on unstable features outside of libcore, liballoc and libkernel. Driver code shouldn't ever rely on unstable features IMO. It is much easier to update only a few core crates than update many drivers as unstable features change. Also it probably shouldn't be overridable by the user, so maybe use `export RUSTC_BOOTSTRAP=...` instead?

Reply:
That is great to know! I think we cannot do it right now due to `allocator_api` (at least), but it could be a very good thing to do as soon as we can, indeed.

Reply:
Related: actually, one thing I would like is to be able to have a "custom automatic prelude" -- I know some people have requested it in the past, but I am not sure how things are.

i.e. I would like to put `#[no_std]` and those `#[features()`] in such a "custom automatic prelude", so that drivers do not even know it is there nor need to care, nor needs to be repeated in every single driver.

Reply:
There is the nightly `#[prelude_import] use kernel::prelude::*;`, but it still requires you to write it in every crate. It just saves you from doing this in every module.

Reply:
I see. Oh well :(

Reply:
Opened https://github.com/Rust-for-Linux/linux/issues/404 to keep track.